packages/sakuli-legacy/src/context/common/test-case/test-case.class.spec.ts

```diff
@@ -241,6 +241,25 @@ describe("TestCase", () => {
             });
         });
 
+        it("should take a screenshot and save it hierarchical when not explicitly specified in props", async () => {
+            // GIVEN
+            const testFolder = "testCaseFolder";
+            const legacyProps = new LegacyProjectProperties();
+            legacyProps.screenshotDir = tmpdir();
+            project = mockPartial<Project>({
+                objectFactory: jest.fn().mockReturnValue(legacyProps)
+            });
+            const SUT = createTestCaseClass(testExecutionContext, project, testFolder);
+            const tc = new SUT("testId", 0, 0);
+            const testError = new Error("testError");
+
+            // WHEN
+            await tc.handleException(testError);
```

Comment:
This code fragment duplicates Line 222-234 now. This implies the test you wrote is already existing but missing a condition. I would suggest to add the `expect(ScreenApi.takeScreenshotWithTimestamp).toBeCalledWith(`${tmpdir()}/UNKNOWN_TESTSUITE_testcase_1/error_UNKNOWN_TESTSUITE_testcase_1`);` condition to the upper test case and alter the name instead of duplicating test code.